src/wasm2js.h

```diff
@@ -2856,9 +2856,14 @@ void Wasm2JSGlue::emitSpecialSupport() {
     } else if (import->base == ABI::wasm2js::ATOMIC_WAIT_I32) {
       out << R"(
   function wasm2js_atomic_wait_i32(ptr, expected, timeoutLow, timeoutHigh) {
-    if (timeoutLow != -1 || timeoutHigh != -1) throw 'unsupported timeout';
+    var timeout = Infinity;
+    if (timeoutHigh >= 0) {
+      // Convert from nanoseconds to milliseconds
+      // Taken from convertI32PairToI53 in emscripten's library_int53.js
+      timeout = ((timeoutLow / 1e6) >>> 0) + timeoutHigh * (4294967296 / 1e6);
```

Comment:
I think maybe `(timeoutLow / 1e6) >>> 0` should be `(timeoutLow >>> 0) / 1e6`. That would allow timeouts of less than 1ms as the current code rounds the output.

Reply:
I thought of that too at first, but it looks like the timeout cannot be sub-millisecond according to the specification. `timeout` is rational (`ℝ`) value:

> Let q be ? ToNumber(timeout).
> 8. If q is NaN or +∞𝔽, let t be +∞; else if q is -∞𝔽, let t be 0; else let t be max(ℝ(q), 0).

Reply:
Hmm, maybe this differs between countries, but isn't ℝ the set of real numbers, not rational? See wikipedia, `The rational numbers (Q) are included in the real numbers (R)` ([link](https://en.wikipedia.org/wiki/Rational_number))

JS does a ToNumber, so it converts to a double, and sub-millisecond values are allowed IIUC

Reply:
> Ah, right! I got confused with natural (ℕ).